src/authenticationinapp/incrementaldecoder.h

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef INCREMENTALDECODER_H
+#define INCREMENTALDECODER_H
+
+#include <QObject>
+
+class QTextStream;
+
+// This class implements the decoder part of this
+// https://en.wikipedia.org/wiki/Incremental_encoding
+
+class IncrementalDecoder final : public QObject {
```

Comment:
It doesn't seem like this needs to be a QObject if all it's doing is wrapping up a file reader.

Reply:
I know, but, if I don't make it a QObject, I cannot use Q_ENUM and I cannot use the types with QTest.